### PR TITLE
Add interactive plot

### PR DESCRIPTION
**Pull request by catrionamurray**
This relies on the .to_df() and .to_nparray() methods in my previous pull request!!
 (not quite sure how that works with the git pull/merge - I think making this pull request would make my previous one redundant?)

Changes in this pull request:
- Added **visualizations/interactive.py** containing the altair code to generate and display an interactive wavelength-averaged plot (I kept _import altair_ out of imports for now). This has the argument _quantity_ which allows the user to also plot the uncertainty
- Added _imshow_interact_ to the **visualizations/__init__.py**
- Added _imshow_interact_ to the **rainbow.py** plotting actions
- Updated version to v0.0.25

Basic usage:
For unbinned data (will throw a warning because it will work but it's slow, also demonstrates the use of the _ylog_ argument):
Screenshot 2022-05-26 at 14 40 11
For binned data (demonstrating changing the _timeformat_ argument to days)
Screenshot 2022-05-26 at 14 40 44

### CODE REVIEW

**Review by zkbt — COMMENTED** [review]
This is *awesome*, @catrionamurray ! I'm so glad this kind of interaction will be so easy now! Thank you! 

A couple of tiny comments:
- Could we please include an example of this with the visualizations section on the documentation? When you get a moment, could you please add a description and example to that notebook? (One subtlety of `mkdocs` that I don't think we've talked about is that I think it's best to commit notebooks with all of the outputs cleared out of them (e.g. run "Restart and Clear Output" just before saving). Otherwise, we'll be saving every version of every plot every time in the repository, which could easily make it grow to an unwieldy size.
- How do we want to handle uncertainties in here? If the signal-to-noise is constant across all wavelengths, then your live average is exactly right. But, if the uncertainties are very different across different wavelengths, then having uncertainty-weighted binning would become more important (= most real datasets). What's your sense? Is this possible in `altair`, or too complicated? I think this is less urgent than your chromatic fitting stuff, so if it's complicated maybe let's just flag it in the documentation for now and return to it later?

File: chromatic/rainbows/visualizations/interactive.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+from ...imports import *
+
+# don't make Altair a necessary part of chromatic
+try:
+    import altair as alt
+    alt.data_transformers.disable_max_rows()
+except Exception as e:
+    print(e)
+    warnings.warn("Issue importing Altair, cannot make interactive plot :(!")
```
[the remark this answers is not in the document]
Comment (catrionamurray) [reply chromatic/rainbows/visualizations/interactive.py:9]: Added installation instructions with pip!

**Comment by catrionamurray** [comment]
I agree an uncertainty-weighted average would be better, I'll investigate whether it's possible in Altair! It's definitely not a built-in function but there may be a way.

**Comment by zkbt** [comment]
This is fabulous! I'll merge it in now, and then we can open a new (very not urgent) issue to talk about the uncertainty-weighting question.

**Comment by zkbt** [comment]
Fantastic work on this, @catrionamurray ! After looking more closely at the `altair` dependencies, I really think there's nothing particularly troublesome associated with it, so I just made it a core dependency for all of `chromatic`. I think there will be a lot of folks who will want to use your snazzy tool, so we might as well include it for everyone!

(And please forgive the working on a holiday; I'm skipping most of Friday this week, so trying to get ahead of things.)

**Pull request merged**